lib/robot.js

```diff
@@ -90,22 +109,15 @@ class Robot {
       console.warn(caller);
     }
 
-    const [name, action] = event.split('.');
-
-    return this.events.on(name, async event => {
-      if (!action || action === event.payload.action) {
-        try {
-          const github = await this.auth(event.payload.installation.id);
-          const context = new Context(event, github);
-          await callback(context, context /* DEPRECATED: for backward compat */);
-        } catch (err) {
-          this.log.error({err, event});
-          if (!this.catchErrors) {
-            throw err;
-          }
-        }
-      }
-    });
+    if (event.constructor === Array) {
+      event.forEach(singleEvent => {
+        this.addEventToListeners(singleEvent, callback);
+      });
+    } else if (event.constructor === String) {
+      this.addEventToListeners(event, callback);
+    } else {
+      console.warn(`robot.on takes a string or array as a first argument, You supplied ${typeof event}`);
```

Comment:
I would either not worry about this use case for now, or raise an error. My preference would be to just not worry about it.